src/Providers/Models/DTO/ModelConfig.php

```diff
@@ -913,6 +944,131 @@ static function (FunctionDeclaration $function_declaration): array {
         return $data;
     }
 
+    /**
+     * Converts the model configuration to required options.
+     *
+     * @since n.e.x.t
+     *
+     * @return list<RequiredOption> The required options.
+     */
+    public function toRequiredOptions(): array
+    {
+        $requiredOptions = [];
+
+        // Map properties that have corresponding OptionEnum values
+        if ($this->outputModalities !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::outputModalities()->value,
+                $this->outputModalities
+            );
+        }
+
+        if ($this->systemInstruction !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::systemInstruction()->value,
+                $this->systemInstruction
+            );
+        }
+
+        if ($this->candidateCount !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::candidateCount()->value,
+                $this->candidateCount
+            );
+        }
+
+        if ($this->maxTokens !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::maxTokens()->value,
+                $this->maxTokens
+            );
+        }
+
+        if ($this->temperature !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::temperature()->value,
+                $this->temperature
+            );
+        }
+
+        if ($this->topP !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::topP()->value,
+                $this->topP
+            );
+        }
+
+        if ($this->topK !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::topK()->value,
+                $this->topK
+            );
+        }
+
+        if ($this->outputMimeType !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::outputMimeType()->value,
+                $this->outputMimeType
+            );
+        }
+
+        if ($this->outputSchema !== null) {
+            $requiredOptions[] = new RequiredOption(
+                OptionEnum::outputSchema()->value,
+                true // Just indicate that schema is required
```

Comment:
Why? I think we can just use `$this->outputSchema` here, just like elsewhere.
```suggestion
                $this->outputSchema
```

Reply:
Resolved in d976ccf